### PR TITLE
Update DEVELOPERS.md documentation to use poetry

### PR DESCRIPTION
**Pull request by yenaing-oo**
Closes https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/issues/268

- Included instructions on setting up virtual environment and installing poetry before installing relevant dependencies with defined extras

### CODE REVIEW

**Comment by yenaing-oo** [comment]
@danloveg 

Could you test that these instructions work as expected on a fresh work environment? I have multiple Python versions installed on my machine, so I just wanted to double-check that there aren't any issues.

File: DEVELOPERS.md [hunk 1]
```diff
@@ -92,12 +92,31 @@ podman-compose -f compose.dev.yml exec app npm run watch
...
+.\env\Scripts\Activate.ps1
+pip install poetry
+```
+
+On MacOS/Linux:
+```shell
+python3 -m venv env
+source env/bin/activate
+pip install poetry
+```
```
Comment (danloveg) [inline DEVELOPERS.md:110]: In the [Poetry installation documentation](https://python-poetry.org/docs/#installation), there's a big warning that poetry needs to be installed in its own environment, separate from the dependencies it manages. The reasoning is that `poetry install` might install an incompatible version of a package that it itself depends on.

The recommended method to install Poetry is with `pipx`.

I would recommend changing this documentation to:

---

# Virtual Environment and Poetry Setup

Python dependencies are managed with [Poetry](https://python-poetry.org/). To install Poetry and create a virtual environment with the minimally necessary packages, follow these instructions:

First, install `pipx` via the [official pipx installation instructions](https://pipx.pypa.io/stable/installation/).

After installing `pipx`, install [Poetry](https://python-poetry.org/):

```shell
pipx install poetry
```

Next, create an activate a virtual environment from the root of the repository:

```shell
# If on Windows:
python -m venv env
.\env\Scripts\Activate.ps1

# If on MacOS or Linux
python3 -m venv env
source env/bin/activate
```

Finally, install the dependencies with `poetry`:

```shell
poetry install
```

**Review by danloveg — CHANGES_REQUESTED** [review]
I recommend installing poetry via `pipx` as a safer method of installing Poetry

File: DEVELOPERS.md (hunk 1 above)
Comment (yenaing-oo) [reply DEVELOPERS.md]: > In the [Poetry installation documentation](https://python-poetry.org/docs/#installation), there's a big warning that poetry needs to be installed in its own environment, separate from the dependencies it manages. The reasoning is that poetry install might install an incompatible version of a package that it itself depends on.

Ah I misunderstood it for "Install everything in a dedicated virtual environment". This makes a lot of sense.

Resolved by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/269/commits/efcda9b6e8aeaca65c9c51fd3bfdd16774362412

**Review by danloveg — APPROVED** [review]

**Pull request merged**